src/metabase/task/search_index.clj

```diff
@@ -74,26 +74,23 @@
 (defn- update-index! []
   (when (search/supports-index?)
     (log/info "Starting Realtime Search Index Update worker")
-    (while true
-      (try
-        (let [batch    (search/get-next-batch! Long/MAX_VALUE 100)
-              _        (log/trace "Processing batch" batch)
-              timer    (u/start-timer)
-              report   (search/bulk-ingest! batch)
-              duration (u/since-ms timer)]
-          (when (seq report)
-            (report->prometheus! duration report)
-            (log/debugf "Indexed search entries in %.0fms %s" duration (sort-by (comp - val) report))))
-        (catch Exception e
-          (prometheus/inc! :metabase-search/index-error)
-          (throw e))))))
-
-(defn- force-scheduled-task! [^JobDetail job ^Trigger trigger]
-  ;; For some reason, using the schedule-task! with a non-durable job causes it to only fire on the first trigger.
-  #_(task/schedule-task! job trigger)
-  (task/delete-task! (.getKey job) (.getKey trigger))
-  (task/add-job! job)
-  (task/add-trigger! trigger))
+    (try
+      (while true
+        (try
+          (let [batch    (search/get-next-batch! Long/MAX_VALUE 100)
+                _        (log/trace "Processing batch" batch)
+                timer    (u/start-timer)
+                report   (search/bulk-ingest! batch)
+                duration (u/since-ms timer)]
+            (when (seq report)
+              (report->prometheus! duration report)
+              (log/debugf "Indexed search entries in %.0fms %s" duration (sort-by (comp - val) report))))
+          (catch Exception e
+            (prometheus/inc! :metabase-search/index-error)
+            (throw e))))
+      (catch Exception e
+        (log/error e "Updating search index failed")
+        (throw (JobExecutionException. "Updating search index failed" e true))))))
```

Comment:
The last argument indicates to immediately retry the job.

Reply:
Discover something new every day. 🙄

Reply:
I feel like this would make some sense to comment it a little bit or extract it in a helper like `do-not-destroy-job-on-exceptions`?